### DISCUSSION TITLE
Project status?

### DISCUSSION THREAD

**Discussion by leroyvn** [xarray-contrib/pint-xarray#188] [General]
Hi all,

First of all, congratulations for this very nice utility package, it works like a charm!

I would like to use it in my software but I still see in the documentation a warning mentioning the experimental status of this project. What should I expect in terms of API stability? Are you aware of critical issues which would prevent usage in production?

Thanks,
Vincent

**Comment by TomNicholas** [comment]
Hi @leroyvn 

> First of all, congratulations for this very nice utility package, it works like a charm!

That's great to hear!

> I would like to use it in my software but I still see in the documentation a warning mentioning the experimental status of this project. What should I expect in terms of API stability? Are you aware of critical issues which would prevent usage in production?

I think this project is beyond merely "experimental" now, but are a couple of things which might change in the API in future. In particular https://github.com/xarray-contrib/pint-xarray/issues/151 and https://github.com/xarray-contrib/pint-xarray/issues/162 might require public changes.

> Are you aware of critical issues which would prevent usage in production?

I don't think these should prevent usage though.

**Reply by keewis** [reply]
in #144 we're planning to use the new exception groups which I think would be a breaking change that we cannot have a deprecation cycle for (but I might be wrong).

By contrast, I think we might be able to have a smooth transition for the changes in #162 (i.e. make the methods on `.pint` deprecated aliases of the xarray methods that work with the new index)

**Reply by leroyvn** [reply]
Thank you for the clarifications, it's definitely a green light for my use case.
